06_puppy/madhavansu/main.go

```diff
@@ -0,0 +1,124 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"sync"
+)
+
+var out io.Writer = os.Stdout
+
+/**
+* Storer implementation in below scenarios
+* Dynamic Map ID: MapStore implementation of Storer backed by a map
+* Static Map ID: SyncStore implementation of Storer backed by a sync.Map
+ */
+//Sync store
+type syncStore struct {
+	sync.Mutex
+	sync.Map
+}
+
+func newSyncStore() *syncStore {
+	return &syncStore{}
+}
+
+func (s *syncStore) createPuppy(in Puppy) uint {
+	s.Lock()
+	defer s.Unlock()
+	s.Store(in.id, in)
+	return in.id
+}
+
+func (s *syncStore) readPuppy(id uint) Puppy {
+	pd, ok := s.Load(id)
+	if !ok {
+		return Puppy{}
+	}
+	p, _ := pd.(Puppy)
+	return p
+}
+
+func (s *syncStore) updatePuppy(id uint, in Puppy) {
+	s.Store(in.id, in)
+}
+
+func (s *syncStore) deletePuppy(id uint) {
+	s.Lock()
+	defer s.Unlock()
+	s.Delete(id)
+}
+
+// Map Store
+type mapStore struct {
+	ms    map[uint]Puppy
+	mapID uint
+}
+
+func newMapStore() *mapStore {
+	return &mapStore{ms: make(map[uint]Puppy)}
+}
+
+func (m *mapStore) createPuppy(in Puppy) uint {
+	m.mapID++
+	in.id = m.mapID
+	m.ms[in.id] = in
+	return m.mapID
+}
+
+func (m *mapStore) readPuppy(id uint) Puppy {
+	p, ok := m.ms[id]
+	if !ok {
+		return Puppy{}
+	}
+	return p
+}
+
+func (m *mapStore) updatePuppy(id uint, in Puppy) {
+	m.ms[id] = in
+}
+
+func (m *mapStore) deletePuppy(id uint) {
+	_, ok := m.ms[id]
+	if ok {
+		delete(m.ms, id)
```

Comment:
`delete` works whether or not something was actually deleted. You don't need to check first.

Reply:
Updated.